crates/node/src/lib.rs

```diff
@@ -134,6 +137,55 @@ impl Shell {
             }
             Request::FinalizeBlock(finalize) => {
                 tracing::debug!("Request FinalizeBlock");
+
```

Comment:
let's put this new block into e.g. `self.try_recheck_process_proposal(&finalize)?;`

Reply:
Done in fc3241dfd11cebfdbc68cb318f3e9b1469d1fcf7